Improve e2e group 8 (RPC debug_trace) to reduce the failing rate

# Rationale
The highest rate of false failing of the e2e groups are:

group 10: 42%: already fixed  #2748
group 8: 12%
group 9: 10%
group 5: 10%
So, the next most important is group is 8 (RPC debug_trace)

There are 6 fails in the total of 12 PR analized:
* https://github.com/0xPolygonHermez/zkevm-node/pull/2730: 1 fail  error: index out of range
* https://github.com/0xPolygonHermez/zkevm-node/pull/2729: 1 fail error: index out of range
* https://github.com/0xPolygonHermez/zkevm-node/pull/2725: 1 fail error: index out of range
* https://github.com/0xPolygonHermez/zkevm-node/pull/2724: 1 fail error: fail to connect executor
* https://github.com/0xPolygonHermez/zkevm-node/pull/2713: 2 fail error: index out of range
# What cause the error?

### runtime error: index out of range [1] with length 1 [recovered]
* https://github.com/0xPolygonHermez/zkevm-node/actions/runs/6655517721/job/18087108962 (running time: 13m 47s) `sc deployment by hash`
* https://github.com/0xPolygonHermez/zkevm-node/actions/runs/6655517721/job/18085956444 (running time: 14m 27s) `sc deployment by hash`
* https://github.com/0xPolygonHermez/zkevm-node/actions/runs/6690513097/job/18175982106 (running time: 14m18s) `eth transfer by hash`
* https://github.com/0xPolygonHermez/zkevm-node/actions/runs/6691578512/job/18179045030 (running time 12m 29s) `erc20 transfer by number`
* https://github.com/0xPolygonHermez/zkevm-node/actions/runs/6692120682/job/18180666975 (running time 14m 14s) ` sc deployment by hash`
```
023-10-26T15:03:39.458Z	DEBUG	e2e-group8/debug_calltracer_test.go:374	tx sent: 0x5b01ecd1c4fd80b31c7e786bff45392a9b7a5c8599b046c6be8b4cdc08b5b836	{"pid": 24077, "version": "v0.1.0"}
2023-10-26T15:03:40.462Z	DEBUG	operations/wait.go:99	Transaction successfully mined: 0x5b01ecd1c4fd80b31c7e786bff45392a9b7a5c8599b046c6be8b4cdc08b5b836	{"pid": 24077, "version": "v0.1.0"}
--- FAIL: TestDebugTraceBlockCallTracer (69.28s)
    --- PASS: TestDebugTraceBlockCallTracer/eth_transfer_by_number (3.06s)
    --- PASS: TestDebugTraceBlockCallTracer/sc_deployment_by_number (3.06s)
    --- PASS: TestDebugTraceBlockCallTracer/sc_call_by_number (6.37s)
    --- PASS: TestDebugTraceBlockCallTracer/erc20_transfer_by_number (10.14s)
    --- PASS: TestDebugTraceBlockCallTracer/eth_transfer_by_hash (3.04s)
    --- FAIL: TestDebugTraceBlockCallTracer/sc_deployment_by_hash (3.06s)
panic: runtime error: index out of range [1] with length 1 [recovered]
	panic: runtime error: index out of range [1] with length 1

goroutine 2646 [running]:
testing.tRunner.func1.2({0x1dc57e0, 0xc00024cb70})
	/opt/hostedtoolcache/go/1.21.3/x64/src/testing/testing.go:1545 +0x366
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.21.3/x64/src/testing/testing.go:1548 +0x630
panic({0x1dc57e0?, 0xc00024cb70?})
	/opt/hostedtoolcache/go/1.21.3/x64/src/runtime/panic.go:[920](https://github.com/0xPolygonHermez/zkevm-node/actions/runs/6655517721/job/18087108962#step:6:921) +0x270
github.com/0xPolygonHermez/zkevm-node/ci/e2e-group8.TestDebugTraceBlockCallTracer.func3(0x0?)
	/home/runner/work/zkevm-node/zkevm-node/ci/e2e-group8/debug_calltracer_test.go:420 +0x1285
testing.tRunner(0xc000303520, 0xc00062c780)
	/opt/hostedtoolcache/go/1.21.3/x64/src/testing/testing.go:1595 +0x239
created by testing.(*T).Run in goroutine 1298
	/opt/hostedtoolcache/go/1.21.3/x64/src/testing/testing.go:1648 +0x82b
FAIL	github.com/0xPolygonHermez/zkevm-node/ci/e2e-group8	419.300s
FAIL
```



### executor/client.go:47	fail to dial: context deadline exceeded
* https://github.com/0xPolygonHermez/zkevm-node/actions/runs/6690314613/job/18175393484 (10m 43s)


```
2023-10-30T08:54:15.942Z	FATAL	executor/client.go:47	fail to dial: context deadline exceeded%!(EXTRA string=
/home/runner/work/zkevm-node/zkevm-node/log/log.go:140 github.com/0xPolygonHermez/zkevm-node/log.appendStackTraceMaybeArgs()
/home/runner/work/zkevm-node/zkevm-node/log/log.go:242 github.com/0xPolygonHermez/zkevm-node/log.Fatalf()
/home/runner/work/zkevm-node/zkevm-node/state/runtime/executor/client.go:47 github.com/0xPolygonHermez/zkevm-node/state/runtime/executor.NewExecutorClient()
/home/runner/work/zkevm-node/zkevm-node/test/operations/manager.go:467 github.com/0xPolygonHermez/zkevm-node/test/operations.initState()
/home/runner/work/zkevm-node/zkevm-node/test/operations/manager.go:103 github.com/0xPolygonHermez/zkevm-node/test/operations.NewManager()
/home/runner/work/zkevm-node/zkevm-node/ci/e2e-group8/debug_calltracer_test.go:43 github.com/0xPolygonHermez/zkevm-node/ci/e2e-group8.TestDebugTraceTransactionCallTracer()
/opt/hostedtoolcache/go/1.21.3/x64/src/testing/testing.go:1595 testing.tRunner()
)	{"pid": 24124, "version": "v0.1.0"}
github.com/0xPolygonHermez/zkevm-node/state/runtime/executor.NewExecutorClient
	/home/runner/work/zkevm-node/zkevm-node/state/runtime/executor/client.go:47
github.com/0xPolygonHermez/zkevm-node/test/operations.initState
	/home/runner/work/zkevm-node/zkevm-node/test/operations/manager.go:467
github.com/0xPolygonHermez/zkevm-node/test/operations.NewManager
	/home/runner/work/zkevm-node/zkevm-node/test/operations/manager.go:103
github.com/0xPolygonHermez/zkevm-node/ci/e2e-group8.TestDebugTraceTransactionCallTracer
	/home/runner/work/zkevm-node/zkevm-node/ci/e2e-group8/debug_calltracer_test.go:43
testing.tRunner
	/opt/hostedtoolcache/go/1.21.3/x64/src/testing/testing.go:1595
FAIL	github.com/0xPolygonHermez/zkevm-node/ci/e2e-group8	143.372s
FAIL
```

--- a/test/e2e/debug_calltracer_test.go
+++ b/test/e2e/debug_calltracer_test.go
@@ -414,7 +414,12 @@ func TestDebugTraceBlockCallTracer(t *testing.T) {
 				resultTransactions := []interface{}{}
 				err = json.Unmarshal(result, &resultTransactions)
 				require.NoError(t, err)
-
+				log.Debugf("transactions length: L1 : %v / L2 %v", len(referenceTransactions), len(resultTransactions))
+				if len(referenceTransactions) != len(resultTransactions) {
+					log.Debug("difer length L1: ", referenceTransactions)
+					log.Debug("difer length L2: ", resultTransactions)
+					require.Fail(t, "transactions results length doesn't match")
+				}
 				for transactionIndex := range referenceTransactions {
 					referenceTransactionMap := referenceTransactions[transactionIndex].(map[string]interface{})
 					resultTransactionMap := resultTransactions[transactionIndex].(map[string]interface{})

--- a/test/e2e/debug_calltracer_test.go
+++ b/test/e2e/debug_calltracer_test.go
@@ -94,7 +94,7 @@ func TestDebugTraceTransactionCallTracer(t *testing.T) {
 		{name: "eth transfer", createSignedTx: createEthTransferSignedTx},
 		{name: "sc deployment", createSignedTx: createScDeploySignedTx},
 		{name: "sc call", prepare: prepareScCall, createSignedTx: createScCallSignedTx},
-		{name: "erc20 transfer", prepare: prepareERC20TransferNoWaitToBeMined, createSignedTx: createERC20TransferSignedTx},
+		{name: "erc20 transfer", prepare: prepareERC20Transfer, createSignedTx: createERC20TransferSignedTx},
 		{name: "create", prepare: prepareCreate, createSignedTx: createCreateSignedTx},
 		{name: "create2", prepare: prepareCreate, createSignedTx: createCreate2SignedTx},
 		{name: "call", prepare: prepareCalls, createSignedTx: createCallSignedTx},

--- a/test/e2e/debug_calltracer_test.go
+++ b/test/e2e/debug_calltracer_test.go
@@ -355,78 +355,85 @@ func TestDebugTraceBlockCallTracer(t *testing.T) {
 			log.Debug("************************ ", tc.name, " ************************")
 
 			for _, network := range networks {
-				log.Debug("------------------------ ", network.Name, " ------------------------")
-				ethereumClient := operations.MustGetClient(network.URL)
-				auth := operations.MustGetAuth(network.PrivateKey, network.ChainID)
+				for true {
+					log.Debug("------------------------ ", network.Name, " ------------------------")
+					ethereumClient := operations.MustGetClient(network.URL)
+					auth := operations.MustGetAuth(network.PrivateKey, network.ChainID)
 
-				var customData map[string]interface{}
-				if tc.prepare != nil {
-					customData, err = tc.prepare(t, ctx, auth, ethereumClient)
+					var customData map[string]interface{}
+					if tc.prepare != nil {
+						customData, err = tc.prepare(t, ctx, auth, ethereumClient)
+						require.NoError(t, err)
+					}
+
+					signedTx, err := tc.createSignedTx(t, ctx, auth, ethereumClient, customData)
 					require.NoError(t, err)
-				}
 
-				signedTx, err := tc.createSignedTx(t, ctx, auth, ethereumClient, customData)
-				require.NoError(t, err)
-
-				err = ethereumClient.SendTransaction(ctx, signedTx)
-				require.NoError(t, err)
-
-				log.Debugf("tx sent: %v", signedTx.Hash().String())
-
-				err = operations.WaitTxToBeMined(ctx, ethereumClient, signedTx, operations.DefaultTimeoutTxToBeMined)
-				if err != nil && !strings.HasPrefix(err.Error(), "transaction has failed, reason:") {
+					err = ethereumClient.SendTransaction(ctx, signedTx)
 					require.NoError(t, err)
+
+					log.Debugf("tx sent: %v", signedTx.Hash().String())
+
+					err = operations.WaitTxToBeMined(ctx, ethereumClient, signedTx, operations.DefaultTimeoutTxToBeMined)
+					if err != nil && !strings.HasPrefix(err.Error(), "transaction has failed, reason:") {
+						require.NoError(t, err)
+					}
+
+					receipt, err := ethereumClient.TransactionReceipt(ctx, signedTx.Hash())
+					require.NoError(t, err)
+
+					debugOptions := map[string]interface{}{
+						"tracer": "callTracer",
+						"tracerConfig": map[string]interface{}{
+							"onlyTopCall": false,
+							"withLog":     true,
+						},
+					}
+
+					var response types.Response
+					if tc.blockNumberOrHash == "number" {
+						log.Infof("debug_traceBlockByNumber %v", receipt.BlockNumber)
+						response, err = client.JSONRPCCall(network.URL, "debug_traceBlockByNumber", hex.EncodeBig(receipt.BlockNumber), debugOptions)
+					} else {
+						log.Infof("debug_traceBlockByHash %v", receipt.BlockHash.String())
+						response, err = client.JSONRPCCall(network.URL, "debug_traceBlockByHash", receipt.BlockHash.String(), debugOptions)
+					}
+					require.NoError(t, err)
+					require.Nil(t, response.Error)
+					require.NotNil(t, response.Result)
+					log.Debugf("[%s/%s] response:%s", tc.name, network.Name, string(response.Result))
+					results[network.Name] = response.Result
+					referenceTransactions := []interface{}{}
+					err = json.Unmarshal(response.Result, &referenceTransactions)
+					require.NoError(t, err)
+					if len(referenceTransactions) > 1 {
+						log.Debug(response.Result)
+						log.Debugf("transactions length: %v", len(referenceTransactions))
+						panic("transactions length > 1")
+					}
 				}
-
-				receipt, err := ethereumClient.TransactionReceipt(ctx, signedTx.Hash())
-				require.NoError(t, err)
-
-				debugOptions := map[string]interface{}{
-					"tracer": "callTracer",
-					"tracerConfig": map[string]interface{}{
-						"onlyTopCall": false,
-						"withLog":     true,
-					},
-				}
-
-				var response types.Response
-				if tc.blockNumberOrHash == "number" {
-					response, err = client.JSONRPCCall(network.URL, "debug_traceBlockByNumber", hex.EncodeBig(receipt.BlockNumber), debugOptions)
-				} else {
-					response, err = client.JSONRPCCall(network.URL, "debug_traceBlockByHash", receipt.BlockHash.String(), debugOptions)
-				}
-				require.NoError(t, err)
-				require.Nil(t, response.Error)
-				require.NotNil(t, response.Result)
-
-				results[network.Name] = response.Result
 			}
 
 			referenceTransactions := []interface{}{}
 			err = json.Unmarshal(results[l1NetworkName], &referenceTransactions)
 			require.NoError(t, err)
 
-			for networkName, result := range results {
-				if networkName == l1NetworkName {
-					continue
-				}
-
-				resultTransactions := []interface{}{}
-				err = json.Unmarshal(result, &resultTransactions)
-				require.NoError(t, err)
-				log.Debugf("transactions length: L1 : %v / L2 %v", len(referenceTransactions), len(resultTransactions))
-				if len(referenceTransactions) != len(resultTransactions) {
-					log.Debug("difer length L1: ", referenceTransactions)
-					log.Debug("difer length L2: ", resultTransactions)
-					require.Fail(t, "transactions results length doesn't match")
-				}
-				for transactionIndex := range referenceTransactions {
-					referenceTransactionMap := referenceTransactions[transactionIndex].(map[string]interface{})
-					resultTransactionMap := resultTransactions[transactionIndex].(map[string]interface{})
-
-					compareCallFrame(t, referenceTransactionMap, resultTransactionMap, networkName)
-				}
+			resultTransactions := []interface{}{}
+			err = json.Unmarshal(results[l2NetworkName], &resultTransactions)
+			require.NoError(t, err)
+			log.Debugf("[%s] transactions length: L1 : %v / L2 %v", tc.name, len(referenceTransactions), len(resultTransactions))
+			if len(referenceTransactions) != len(resultTransactions) {
+				log.Debug("[%s] difer length L1: ", tc.name, results[l1NetworkName])
+				log.Debug("[%s] difer length L2: ", tc.name, results[l2NetworkName])
+				require.Fail(t, "transactions results length doesn't match")
 			}
+			for transactionIndex := range referenceTransactions {
+				referenceTransactionMap := referenceTransactions[transactionIndex].(map[string]interface{})
+				resultTransactionMap := resultTransactions[transactionIndex].(map[string]interface{})
+
+				compareCallFrame(t, referenceTransactionMap, resultTransactionMap, l2NetworkName)
+			}
+
 		})
 	}
 }

--- a/test/e2e/debug_calltracer_test.go
+++ b/test/e2e/debug_calltracer_test.go
@@ -21,6 +21,10 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+const (
+	dockersArePreLaunchedForCallTracerTests = false
+)
+
 func TestDebugTraceTransactionCallTracer(t *testing.T) {
 	if testing.Short() {
 		t.Skip()
@@ -30,23 +34,30 @@ func TestDebugTraceTransactionCallTracer(t *testing.T) {
 	const l2ExplorerRPCComponentName = "l2-explorer-json-rpc"
 
 	var err error
-	err = operations.Teardown()
-	require.NoError(t, err)
-
-	defer func() {
-		require.NoError(t, operations.Teardown())
-		require.NoError(t, operations.StopComponent(l2ExplorerRPCComponentName))
-	}()
+	if !dockersArePreLaunchedForCallTracerTests {
+		err = operations.Teardown()
+		require.NoError(t, err)
+		defer func() {
+			require.NoError(t, operations.Teardown())
+			require.NoError(t, operations.StopComponent(l2ExplorerRPCComponentName))
+		}()
+	}
 
 	ctx := context.Background()
 	opsCfg := operations.GetDefaultOperationsConfig()
-	opsMan, err := operations.NewManager(ctx, opsCfg)
-	require.NoError(t, err)
-	err = opsMan.Setup()
-	require.NoError(t, err)
-
-	err = operations.StartComponent(l2ExplorerRPCComponentName, func() (bool, error) { return operations.NodeUpCondition(l2NetworkURL) })
-	require.NoError(t, err)
+	var opsMan *operations.Manager
+	if !dockersArePreLaunchedForCallTracerTests {
+		opsMan, err = operations.NewManager(ctx, opsCfg)
+		require.NoError(t, err)
+		err = opsMan.Setup()
+		require.NoError(t, err)
+		err = operations.StartComponent(l2ExplorerRPCComponentName, func() (bool, error) { return operations.NodeUpCondition(l2NetworkURL) })
+		require.NoError(t, err)
+	} else {
+		log.Info("Using pre-launched dockers: no reset Database")
+		opsMan, err = operations.NewManagerNoInitDB(ctx, opsCfg)
+		require.NoError(t, err)
+	}
 
 	const l1NetworkName, l2NetworkName = "Local L1", "Local L2"
 
@@ -83,7 +94,7 @@ func TestDebugTraceTransactionCallTracer(t *testing.T) {
 		{name: "eth transfer", createSignedTx: createEthTransferSignedTx},
 		{name: "sc deployment", createSignedTx: createScDeploySignedTx},
 		{name: "sc call", prepare: prepareScCall, createSignedTx: createScCallSignedTx},
-		{name: "erc20 transfer", prepare: prepareERC20Transfer, createSignedTx: createERC20TransferSignedTx},
+		{name: "erc20 transfer", prepare: prepareERC20TransferNoWaitToBeMined, createSignedTx: createERC20TransferSignedTx},
 		{name: "create", prepare: prepareCreate, createSignedTx: createCreateSignedTx},
 		{name: "create2", prepare: prepareCreate, createSignedTx: createCreate2SignedTx},
 		{name: "call", prepare: prepareCalls, createSignedTx: createCallSignedTx},
@@ -277,23 +288,30 @@ func TestDebugTraceBlockCallTracer(t *testing.T) {
 	const l2ExplorerRPCComponentName = "l2-explorer-json-rpc"
 
 	var err error
-	err = operations.Teardown()
-	require.NoError(t, err)
-
-	defer func() {
-		require.NoError(t, operations.Teardown())
-		require.NoError(t, operations.StopComponent(l2ExplorerRPCComponentName))
-	}()
+	if !dockersArePreLaunchedForCallTracerTests {
+		err = operations.Teardown()
+		require.NoError(t, err)
+		defer func() {
+			require.NoError(t, operations.Teardown())
+			require.NoError(t, operations.StopComponent(l2ExplorerRPCComponentName))
+		}()
+	}
 
 	ctx := context.Background()
 	opsCfg := operations.GetDefaultOperationsConfig()
-	opsMan, err := operations.NewManager(ctx, opsCfg)
-	require.NoError(t, err)
-	err = opsMan.Setup()
-	require.NoError(t, err)
-
-	err = operations.StartComponent(l2ExplorerRPCComponentName, func() (bool, error) { return operations.NodeUpCondition(l2NetworkURL) })
-	require.NoError(t, err)
+	var opsMan *operations.Manager
+	if !dockersArePreLaunchedForCallTracerTests {
+		opsMan, err = operations.NewManager(ctx, opsCfg)
+		require.NoError(t, err)
+		err = opsMan.Setup()
+		require.NoError(t, err)
+		err = operations.StartComponent(l2ExplorerRPCComponentName, func() (bool, error) { return operations.NodeUpCondition(l2NetworkURL) })
+		require.NoError(t, err)
+	} else {
+		log.Info("Using pre-launched dockers: no reset Database")
+		//opsMan, err = operations.NewManagerNoInitDB(ctx, opsCfg)
+		//require.NoError(t, err)
+	}
 
 	const l1NetworkName, l2NetworkName = "Local L1", "Local L2"
 
@@ -317,8 +335,9 @@ func TestDebugTraceBlockCallTracer(t *testing.T) {
 			PrivateKey: operations.DefaultSequencerPrivateKey,
 		},
 	}
-
-	results := map[string]json.RawMessage{}
+	require.Equal(t, len(networks), 2, "only support 2 networks!")
+	//var results map[string]map[string]interface{}
+	results := map[string]map[string]interface{}{}
 
 	type testCase struct {
 		name              string
@@ -401,30 +420,30 @@ func TestDebugTraceBlockCallTracer(t *testing.T) {
 				require.NoError(t, err)
 				require.Nil(t, response.Error)
 				require.NotNil(t, response.Result)
-				log.Debugf("[%s/%s] response:%s", tc.name, network.Name, string(response.Result))
-				results[network.Name] = response.Result
+				log.Debugf("%s response:%s", debugID, string(response.Result))
+
+				txHash := signedTx.Hash().String()
+				resultForTx := findTxInResponse(t, response.Result, txHash, debugID)
+				results[network.Name] = resultForTx
 			}
 
-			referenceTransactions := []interface{}{}
-			err = json.Unmarshal(results[l1NetworkName], &referenceTransactions)
-			require.NoError(t, err)
-
-			resultTransactions := []interface{}{}
-			err = json.Unmarshal(results[l2NetworkName], &resultTransactions)
-			require.NoError(t, err)
-			log.Debugf("[%s] transactions length: L1: %v / L2: %v", tc.name, len(referenceTransactions), len(resultTransactions))
-			if len(referenceTransactions) != len(resultTransactions) {
-				log.Debug("[%s] difer length L1: ", tc.name, results[l1NetworkName])
-				log.Debug("[%s] difer length L2: ", tc.name, results[l2NetworkName])
-				require.Fail(t, tc.name+"  transactions results length doesn't match")
-			}
-			for transactionIndex := range referenceTransactions {
-				referenceTransactionMap := referenceTransactions[transactionIndex].(map[string]interface{})
-				resultTransactionMap := resultTransactions[transactionIndex].(map[string]interface{})
-
-				compareCallFrame(t, referenceTransactionMap, resultTransactionMap, l2NetworkName)
-			}
-
+			referenceTransactions := results[l1NetworkName]
+			resultTransactions := results[l2NetworkName]
+			compareCallFrame(t, referenceTransactions, resultTransactions, l2NetworkName)
 		})
 	}
+}
+
+func findTxInResponse(t *testing.T, response json.RawMessage, txHash string, debugPrefix string) map[string]interface{} {
+	valueMap := []interface{}{}
+	err := json.Unmarshal(response, &valueMap)
+	require.NoError(t, err)
+	log.Infof("%s Reponse Length: %d", debugPrefix, len(valueMap))
+	for transactionIndex := range valueMap {
+		if valueMap[transactionIndex].(map[string]interface{})["txHash"] == txHash {
+			return valueMap[transactionIndex].(map[string]interface{})
+		}
+	}
+	log.Infof("%s Transaction not found: %s, returning first index", debugPrefix, txHash)
+	return valueMap[0].(map[string]interface{})
 }

--- a/test/e2e/debug_calltracer_test.go
+++ b/test/e2e/debug_calltracer_test.go
@@ -45,9 +45,8 @@ func TestDebugTraceTransactionCallTracer(t *testing.T) {
 
 	ctx := context.Background()
 	opsCfg := operations.GetDefaultOperationsConfig()
-	var opsMan *operations.Manager
 	if !dockersArePreLaunchedForCallTracerTests {
-		opsMan, err = operations.NewManager(ctx, opsCfg)
+		opsMan, err := operations.NewManager(ctx, opsCfg)
 		require.NoError(t, err)
 		err = opsMan.Setup()
 		require.NoError(t, err)
@@ -55,8 +54,6 @@ func TestDebugTraceTransactionCallTracer(t *testing.T) {
 		require.NoError(t, err)
 	} else {
 		log.Info("Using pre-launched dockers: no reset Database")
-		opsMan, err = operations.NewManagerNoInitDB(ctx, opsCfg)
-		require.NoError(t, err)
 	}
 
 	const l1NetworkName, l2NetworkName = "Local L1", "Local L2"
@@ -299,9 +296,8 @@ func TestDebugTraceBlockCallTracer(t *testing.T) {
 
 	ctx := context.Background()
 	opsCfg := operations.GetDefaultOperationsConfig()
-	var opsMan *operations.Manager
 	if !dockersArePreLaunchedForCallTracerTests {
-		opsMan, err = operations.NewManager(ctx, opsCfg)
+		opsMan, err := operations.NewManager(ctx, opsCfg)
 		require.NoError(t, err)
 		err = opsMan.Setup()
 		require.NoError(t, err)
@@ -309,8 +305,6 @@ func TestDebugTraceBlockCallTracer(t *testing.T) {
 		require.NoError(t, err)
 	} else {
 		log.Info("Using pre-launched dockers: no reset Database")
-		//opsMan, err = operations.NewManagerNoInitDB(ctx, opsCfg)
-		//require.NoError(t, err)
 	}
 
 	const l1NetworkName, l2NetworkName = "Local L1", "Local L2"

--- a/test/e2e/debug_calltracer_test.go
+++ b/test/e2e/debug_calltracer_test.go
@@ -355,63 +355,54 @@ func TestDebugTraceBlockCallTracer(t *testing.T) {
 			log.Debug("************************ ", tc.name, " ************************")
 
 			for _, network := range networks {
-				for true {
-					log.Debug("------------------------ ", network.Name, " ------------------------")
-					ethereumClient := operations.MustGetClient(network.URL)
-					auth := operations.MustGetAuth(network.PrivateKey, network.ChainID)
+				debugID := fmt.Sprintf("[%s/%s]", tc.name, network.Name)
+				log.Debug("------------------------ ", network.Name, " ------------------------")
+				ethereumClient := operations.MustGetClient(network.URL)
+				auth := operations.MustGetAuth(network.PrivateKey, network.ChainID)
 
-					var customData map[string]interface{}
-					if tc.prepare != nil {
-						customData, err = tc.prepare(t, ctx, auth, ethereumClient)
-						require.NoError(t, err)
-					}
-
-					signedTx, err := tc.createSignedTx(t, ctx, auth, ethereumClient, customData)
+				var customData map[string]interface{}
+				if tc.prepare != nil {
+					customData, err = tc.prepare(t, ctx, auth, ethereumClient)
 					require.NoError(t, err)
-
-					err = ethereumClient.SendTransaction(ctx, signedTx)
-					require.NoError(t, err)
-
-					log.Debugf("tx sent: %v", signedTx.Hash().String())
-
-					err = operations.WaitTxToBeMined(ctx, ethereumClient, signedTx, operations.DefaultTimeoutTxToBeMined)
-					if err != nil && !strings.HasPrefix(err.Error(), "transaction has failed, reason:") {
-						require.NoError(t, err)
-					}
-
-					receipt, err := ethereumClient.TransactionReceipt(ctx, signedTx.Hash())
-					require.NoError(t, err)
-
-					debugOptions := map[string]interface{}{
-						"tracer": "callTracer",
-						"tracerConfig": map[string]interface{}{
-							"onlyTopCall": false,
-							"withLog":     true,
-						},
-					}
-
-					var response types.Response
-					if tc.blockNumberOrHash == "number" {
-						log.Infof("debug_traceBlockByNumber %v", receipt.BlockNumber)
-						response, err = client.JSONRPCCall(network.URL, "debug_traceBlockByNumber", hex.EncodeBig(receipt.BlockNumber), debugOptions)
-					} else {
-						log.Infof("debug_traceBlockByHash %v", receipt.BlockHash.String())
-						response, err = client.JSONRPCCall(network.URL, "debug_traceBlockByHash", receipt.BlockHash.String(), debugOptions)
-					}
-					require.NoError(t, err)
-					require.Nil(t, response.Error)
-					require.NotNil(t, response.Result)
-					log.Debugf("[%s/%s] response:%s", tc.name, network.Name, string(response.Result))
-					results[network.Name] = response.Result
-					referenceTransactions := []interface{}{}
-					err = json.Unmarshal(response.Result, &referenceTransactions)
-					require.NoError(t, err)
-					if len(referenceTransactions) > 1 {
-						log.Debug(response.Result)
-						log.Debugf("transactions length: %v", len(referenceTransactions))
-						panic("transactions length > 1")
-					}
 				}
+
+				signedTx, err := tc.createSignedTx(t, ctx, auth, ethereumClient, customData)
+				require.NoError(t, err)
+
+				err = ethereumClient.SendTransaction(ctx, signedTx)
+				require.NoError(t, err)
+
+				log.Debugf("%s tx sent: %v", debugID, signedTx.Hash().String())
+
+				err = operations.WaitTxToBeMined(ctx, ethereumClient, signedTx, operations.DefaultTimeoutTxToBeMined)
+				if err != nil && !strings.HasPrefix(err.Error(), "transaction has failed, reason:") {
+					require.NoError(t, err)
+				}
+
+				receipt, err := ethereumClient.TransactionReceipt(ctx, signedTx.Hash())
+				require.NoError(t, err)
+
+				debugOptions := map[string]interface{}{
+					"tracer": "callTracer",
+					"tracerConfig": map[string]interface{}{
+						"onlyTopCall": false,
+						"withLog":     true,
+					},
+				}
+
+				var response types.Response
+				if tc.blockNumberOrHash == "number" {
+					log.Infof("%s debug_traceBlockByNumber %v", debugID, receipt.BlockNumber)
+					response, err = client.JSONRPCCall(network.URL, "debug_traceBlockByNumber", hex.EncodeBig(receipt.BlockNumber), debugOptions)
+				} else {
+					log.Infof("%s debug_traceBlockByHash %v", debugID, receipt.BlockHash.String())
+					response, err = client.JSONRPCCall(network.URL, "debug_traceBlockByHash", receipt.BlockHash.String(), debugOptions)
+				}
+				require.NoError(t, err)
+				require.Nil(t, response.Error)
+				require.NotNil(t, response.Result)
+				log.Debugf("[%s/%s] response:%s", tc.name, network.Name, string(response.Result))
+				results[network.Name] = response.Result
 			}
 
 			referenceTransactions := []interface{}{}
@@ -421,11 +412,11 @@ func TestDebugTraceBlockCallTracer(t *testing.T) {
 			resultTransactions := []interface{}{}
 			err = json.Unmarshal(results[l2NetworkName], &resultTransactions)
 			require.NoError(t, err)
-			log.Debugf("[%s] transactions length: L1 : %v / L2 %v", tc.name, len(referenceTransactions), len(resultTransactions))
+			log.Debugf("[%s] transactions length: L1: %v / L2: %v", tc.name, len(referenceTransactions), len(resultTransactions))
 			if len(referenceTransactions) != len(resultTransactions) {
 				log.Debug("[%s] difer length L1: ", tc.name, results[l1NetworkName])
 				log.Debug("[%s] difer length L2: ", tc.name, results[l2NetworkName])
-				require.Fail(t, "transactions results length doesn't match")
+				require.Fail(t, tc.name+"  transactions results length doesn't match")
 			}
 			for transactionIndex := range referenceTransactions {
 				referenceTransactionMap := referenceTransactions[transactionIndex].(map[string]interface{})

--- a/test/e2e/debug_shared.go
+++ b/test/e2e/debug_shared.go
@@ -10,6 +10,7 @@ import (
 	"testing"
 
 	"github.com/0xPolygonHermez/zkevm-node/hex"
+	"github.com/0xPolygonHermez/zkevm-node/log"
 	"github.com/0xPolygonHermez/zkevm-node/test/contracts/bin/BridgeA"
 	"github.com/0xPolygonHermez/zkevm-node/test/contracts/bin/BridgeB"
 	"github.com/0xPolygonHermez/zkevm-node/test/contracts/bin/BridgeC"
@@ -109,17 +110,29 @@ func createScCallSignedTx(t *testing.T, ctx context.Context, auth *bind.Transact
 }
 
 func prepareERC20Transfer(t *testing.T, ctx context.Context, auth *bind.TransactOpts, client *ethclient.Client) (map[string]interface{}, error) {
+	return prepareERC20TransferInternal(t, ctx, auth, client, true)
+}
+
+func prepareERC20TransferNoWaitToBeMined(t *testing.T, ctx context.Context, auth *bind.TransactOpts, client *ethclient.Client) (map[string]interface{}, error) {
+	return prepareERC20TransferInternal(t, ctx, auth, client, false)
+}
+
+func prepareERC20TransferInternal(t *testing.T, ctx context.Context, auth *bind.TransactOpts, client *ethclient.Client, waitToBeMined bool) (map[string]interface{}, error) {
 	_, tx, sc, err := ERC20.DeployERC20(auth, client, "MyToken", "MT")
 	require.NoError(t, err)
-
-	err = operations.WaitTxToBeMined(ctx, client, tx, operations.DefaultTimeoutTxToBeMined)
-	require.NoError(t, err)
+	log.Debugf("prepareERC20Transfer DeployERC20 tx: %s", tx.Hash().String())
+	if waitToBeMined {
+		err = operations.WaitTxToBeMined(ctx, client, tx, operations.DefaultTimeoutTxToBeMined)
+		require.NoError(t, err)
+	}
 
 	tx, err = sc.Mint(auth, big.NewInt(1000000000))
 	require.NoError(t, err)
-
-	err = operations.WaitTxToBeMined(ctx, client, tx, operations.DefaultTimeoutTxToBeMined)
-	require.NoError(t, err)
+	log.Debugf("prepareERC20Transfer Mint tx: %s", tx.Hash().String())
+	if waitToBeMined {
+		err = operations.WaitTxToBeMined(ctx, client, tx, operations.DefaultTimeoutTxToBeMined)
+		require.NoError(t, err)
+	}
 
 	return map[string]interface{}{
 		"sc": sc,

--- a/test/e2e/debug_shared.go
+++ b/test/e2e/debug_shared.go
@@ -110,29 +110,17 @@ func createScCallSignedTx(t *testing.T, ctx context.Context, auth *bind.Transact
 }
 
 func prepareERC20Transfer(t *testing.T, ctx context.Context, auth *bind.TransactOpts, client *ethclient.Client) (map[string]interface{}, error) {
-	return prepareERC20TransferInternal(t, ctx, auth, client, true)
-}
-
-func prepareERC20TransferNoWaitToBeMined(t *testing.T, ctx context.Context, auth *bind.TransactOpts, client *ethclient.Client) (map[string]interface{}, error) {
-	return prepareERC20TransferInternal(t, ctx, auth, client, false)
-}
-
-func prepareERC20TransferInternal(t *testing.T, ctx context.Context, auth *bind.TransactOpts, client *ethclient.Client, waitToBeMined bool) (map[string]interface{}, error) {
 	_, tx, sc, err := ERC20.DeployERC20(auth, client, "MyToken", "MT")
 	require.NoError(t, err)
 	log.Debugf("prepareERC20Transfer DeployERC20 tx: %s", tx.Hash().String())
-	if waitToBeMined {
-		err = operations.WaitTxToBeMined(ctx, client, tx, operations.DefaultTimeoutTxToBeMined)
-		require.NoError(t, err)
-	}
+	err = operations.WaitTxToBeMined(ctx, client, tx, operations.DefaultTimeoutTxToBeMined)
+	require.NoError(t, err)
 
 	tx, err = sc.Mint(auth, big.NewInt(1000000000))
 	require.NoError(t, err)
 	log.Debugf("prepareERC20Transfer Mint tx: %s", tx.Hash().String())
-	if waitToBeMined {
-		err = operations.WaitTxToBeMined(ctx, client, tx, operations.DefaultTimeoutTxToBeMined)
-		require.NoError(t, err)
-	}
+	err = operations.WaitTxToBeMined(ctx, client, tx, operations.DefaultTimeoutTxToBeMined)
+	require.NoError(t, err)
 
 	return map[string]interface{}{
 		"sc": sc,

